### PR TITLE
Fix tests after merging #6147

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

This addresses static analysis issues reported after merging #6147. It also addresses some of the feedback in https://github.com/ampproject/amp-wp/pull/6147#pullrequestreview-753206599.

See #5939.

## Checklist

- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: src/Admin/SupportLink.php [hunk 1]
```diff
@@ -60,12 +60,9 @@ public function register() {
...
 	 * @return void
 	 */
-	public function admin_bar_menu( $wp_admin_bar ) {
+	public function admin_bar_menu( WP_Admin_Bar $wp_admin_bar ) {
 
-		if ( empty( $wp_admin_bar ) ||
-			! is_a( $wp_admin_bar, 'WP_Admin_Bar' ) ||
-			! array_key_exists( 'amp', $wp_admin_bar->get_nodes() )
-		) {
+		if ( array_key_exists( 'amp', $wp_admin_bar->get_nodes() ) ) {
```
Comment (westonruter) [inline src/Admin/SupportLink.php]: Oops:

```suggestion
		if ( ! array_key_exists( 'amp', $wp_admin_bar->get_nodes() ) ) {
```

Comment (westonruter) [reply src/Admin/SupportLink.php]: Also, this can be made more efficient via:

```suggestion
		if ( ! $wp_admin_bar->get_node( 'amp' ) ) {
```

**Comment by westonruter** [comment]
Looks like E2E needs some love. cc @delawski

**Comment by pierlon** [comment]
Seems to be another flaky E2E test, seeing how best it can be fixed.

**Comment by pierlon** [comment]
The E2E tests are passing consistently locally for me so I'm not sure what the problem is. Maybe this was a one-off error?

**Comment by westonruter** [comment]
Huh, and they're passing on GHA now too. Last I checked they were failing. Did you restart the workflow?

**Comment by pierlon** [comment]
Oh yep just wanted to confirm if it failed after another run.

**Review by pierlon — APPROVED** [review]
LGTM.

**Comment by westonruter** [comment]
There seem to be a lot of one-off errors 😄

**Pull request merged**